include/xgboost/data.h

```diff
@@ -559,8 +559,9 @@ class DMatrix {
    *
    * \param uri The URI of input.
    * \param silent Whether print information during loading.
-   * \param data_split_mode In distributed mode, split the input according this mode; otherwise,
-   *                        it's just an indicator on how the input was split beforehand.
+   * \param data_split_mode In distributed mode, if the data split mode is row, split the input by
```

Comment:
Let's remove the row splitting as well. I can help with that if it doesn't conflict with your PR.

Reply:
Are you sure that's what we want to do? In the text readers there is logic to optimize for reading a portion of the file, so in terms of performance it's probably that far off from reading a whole file. This behavior has been around since the beginning of xgboost, I wonder how many people actually rely on it.

Reply:
Done.

Reply:
Thank you for removing the old code, could you please help fix the lint errors?

Reply:
I will fix it in a separated PR for backporting: https://github.com/dmlc/xgboost/pull/9634 .